packages/app/src/applications/Application.tsx

```diff
@@ -403,14 +404,27 @@ class ApplicationImpl extends React.PureComponent {
     const useNativeWindowOpen = !this.props.notUseNativeWindowOpen;
     const tabUrl = tab.get('url', '');
     const {
-      applicationId, applicationName, applicationIcon, themeColor, manifestURL,
+      applicationId, applicationName, applicationIcon, 
+      appstoreApplicationId, themeColor, manifestURL,
       askResetApplication, onChooseAccount,
       crashed, errorCode, errorDescription,
       canGoBack, themeGradient, email,
       promptBasicAuth, performBasicAuth, basicAuthInfo,
       useDefaultSession,
     } = this.props;
 
+    const partition = useDefaultSession ? '' : `persist:${applicationId}`;
+
+    // disable SSL check for private applications
+    if (!useDefaultSession && Number(appstoreApplicationId) > 1000000) {
+      services.defaultSession.disableSslCertVerification(partition);
+
+      // const appSession = remote.session.fromPartition(partition);
+      // appSession.setCertificateVerifyProc((_, callback) => {
+      //   callback(0);
+      // })
+    }
```

Comment:
Hey @magne4000 , need your help here.
The issue could be fixed with 3 lines of code using `@electrone/remote`, but I think we should avoid to use it.
So I had tried to implement new method in `SessionService` and stuck 😞

Reply:
`SessionServiceImpl` was created at some point with the hope that it would be reusable with partitions, but clearly it's not the case.

What I would suggest would be to:
- simplify `SessionServiceImpl` constructor and `initSession` by removing `SessionOptions` and remove handling of partitions
- rename others existing functions to `getDefaultUserAgent`, `getDefaultCookies` and `getDefaultSession`
- move `disableSslCertVerification` to `SessionServiceImpl`, and make it so that it takes a partition as argument of the function

So instead of representing a particular session `SessionService` now represent a way to access all of them. Then we should finally rename all `service.defaultSession` to `service.session`.

What do you think?

Reply:
We are calling from renderer to main here. We have to use ipc.
That's the problem I struggle to. 😀

Reply:
I pushed a fix.
Generally, the way to think when implementing or using Services should be:

- Find the proper way to dispatch a message in the store:
  - From a renderer, use redux store (our case here)
  - From a worker, provide store in ServiceImpl init, then use it
  - From main, provide a Worker Service in init (i.e. a Service whose impl runs in Worker), then use it (should be last resort, as its a complex pattern and difficult to debug)
- Then, no need to store the message in a state if its just a simple message passing
- Then, in a saga, call the right service

Reply:
That's a really complicated thing for me.
Thank you.